app/src/main/java/org/eyeseetea/malariacare/presenter/ReviewPresenter.java

```diff
@@ -0,0 +1,122 @@
+import org.eyeseetea.malariacare.utils.Constants;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class ReviewPresenter implements ReviewScreenAdapter.onClickListener {
+
+    public interface ReviewView {
+        void showValues(List<Value> values);
+
+        void initListView();
+    }
+
+    ReviewView view;
+
+    public ReviewPresenter() {
+    }
+
+    public void attachView(ReviewView reviewView) {
+        this.view = reviewView;
+        view.showValues(prepareValues());
+        view.initListView();
+    }
+
+    private List<org.eyeseetea.malariacare.domain.entity.Value> prepareValues() {
+        Iterator<String> colorIterator;
+        List<org.eyeseetea.malariacare.domain.entity.Value> preparedValues = new ArrayList<>();
+        List<org.eyeseetea.malariacare.data.database.model.Value> values;
+        values = getReviewValues();
+        values = orderValues(values);
+        colorIterator = Iterables.cycle(createBackgroundColorList(values)).iterator();
+        for (org.eyeseetea.malariacare.data.database.model.Value value : values) {
+            org.eyeseetea.malariacare.domain.entity.Value preparedValue =
+                    new org.eyeseetea.malariacare.domain.entity.Value(value.getValue());
+            if (value.getQuestion() != null) {
+                preparedValue.setQuestionUId(value.getQuestion().getUid());
+            }
+            if (value.getOption() != null) {
+                preparedValue.setInternationalizedCode(
+                        value.getOption().getInternationalizedCode());
+            }
+            if (colorIterator.hasNext()) {
+                preparedValue.setBackgroundColor(colorIterator.next());
+            }
+            preparedValues.add(preparedValue);
+        }
+        return preparedValues;
+    }
+
+    private List<String> createBackgroundColorList(
+            List<org.eyeseetea.malariacare.data.database.model.Value> values) {
+        List<String> colorsList = new ArrayList<>();
+        for (org.eyeseetea.malariacare.data.database.model.Value value : values) {
+            if (value.getOption() != null && value.getOption().getBackground_colour() != null) {
+                String color = "#" + value.getOption().getBackground_colour();
+                if (!colorsList.contains(color)) {
+                    colorsList.add(color);
+                }
+            }
+        }
+        //Hardcoded colors for a colorList without colors.
+        if (colorsList.size() == 0) {
+            colorsList.add("#4d3a4b");
+        }
+        if (colorsList.size() == 1 && values.size() > 1) {
+            colorsList.add("#9c7f9b");
+        }
+        return colorsList;
+    }
+
+    private List<org.eyeseetea.malariacare.data.database.model.Value> orderValues(
+            List<org.eyeseetea.malariacare.data.database.model.Value> values) {
+        return ReviewFragmentStrategy.orderValues(values);
```

Comment:
Presenter should not invoke directly to presentation object. Only can invoke methods of ReviewView to communicate with presentation layer. I think that order should be domain responsibility, can you refactor this code for order in GetReviewValuesBySurveyIdUseCase and create strategies by variant if its necessary?